lib/matplotlib/colorbar.py

```diff
@@ -852,12 +979,28 @@ def _mesh(self):
             y = self._uniform_y(self._central_N())
         else:
             y = self._proportional_y()
+        # if boundaries and values are None, then we can go ahead and
+        # scale this up for Auto tick location.  Otherwise we
+        # want to keep normalized between 0 and 1 and use manual tick
+        # locations.
+        if self._use_adjustable():
+            y = self.norm.inverse(y)
+            x = self.norm.inverse(x)
+        else:
+            dy = 1.0
         self._y = y
+
         X, Y = np.meshgrid(x, y)
         if self._extend_lower() and not self.extendrect:
-            X[0, :] = 0.5
+            if self._use_adjustable():
+                X[0, :] = self.norm.inverse(0.5)
+            else:
+                X[0, :] = 0.5
         if self._extend_upper() and not self.extendrect:
-            X[-1, :] = 0.5
+            if self._use_adjustable():
+                X[-1, :] = self.norm.inverse(0.5)
```

Comment:
I don't like the business of using a log scale for X when it is used for Y, but it seems to be a necessary hack for the underlying API problem: we have a mechanism for specifying the aspect ratio at draw time in data units, but not for specifying the box aspect ratio directly in screen units.

Reply:
The aspect ratio is part of it, but calculating the extend point (for the pointy parts at the end of the colorbar) was also going to be a bit ugly w/o doing this, I think.  Its possible that something could be done w/ transforms?  I'll admit my ignorance about transforms is somewhat vast.